src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -332,22 +333,65 @@ EclipseGrid::EclipseGrid(const Deck& deck, const int * actnum)
         return this->m_circle;
     }
 
+
     void EclipseGrid::initGrid(const Deck& deck, const int* actnum)
     {
-        if (deck.hasKeyword<ParserKeywords::RADIAL>()) {
+        enum GridType { COORD, DEPTHZ, TOPS, RADIAL, SPIDER, GDFILE, GT_SIZE };
+
+        std::vector<int> found;
+        if (hasCornerPointKeywords(deck)) {
+            found.push_back(COORD);
+        }
+        if (hasDVDEPTHZKeywords(deck)) {
+            found.push_back(DEPTHZ);
+        }
+        if (hasDTOPSKeywords(deck)) {
+            found.push_back(TOPS);
+        }
+        if (hasRadialKeywords(deck)) {
+            found.push_back(RADIAL);
+        }
+        if (hasSpiderKeywords(deck)) {
+            found.push_back(SPIDER);
+        }
+        if (hasGDFILE(deck)) {
+            found.push_back(GDFILE);
+        }
+
+        const std::string messages[GT_SIZE] {
+            "COORD with ZCORN creates a corner-point grid",
+            "DEPTHZ with DXV, DYV, DZV creates a cartesian grid",
+            "TOPS with DX/DXV, DY/DYV, DZ/DZV creates a cartesian grid",
+            "RADIAL with DR/DRV, DTHETA/DTHETAV, DZ/DZV and TOPS creates a cylindrical grid",
+            "SPIDER with DR/DRV, DTHETA/DTHETAV, DZ/DZV and TOPS creates a spider grid",
+            "GDFILE reads a grid from file"};
+
+        if (found.size() == 0) {
```

Comment:
It's usually better to check `found.empty()` instead of `found.size() == 0`.

Reply:
Done.